nixos/modules/services/x11/redshift.nix

```diff
@@ -35,42 +46,41 @@ in {
       '';
     };
 
-    temperature = {
-      day = mkOption {
-        type = types.int;
-        default = 5500;
-        description = ''
-          Colour temperature to use during the day, between
-          <literal>1000</literal> and <literal>25000</literal> K.
-        '';
-      };
-      night = mkOption {
-        type = types.int;
-        default = 3700;
-        description = ''
-          Colour temperature to use at night, between
-          <literal>1000</literal> and <literal>25000</literal> K.
-        '';
-      };
+    settings = mkOption {
+      type = with types; attrsOf (attrsOf (nullOr (oneOf [ str float bool int ])));
+      default = {};
+      description = ''
+        The configuration to pass to redshift.
+        See <command>man redshift</command> under the section
+        CONFIGURATION FILE for options.
+      '';
+      example = literalExample ''{
+        redshift = {
+          dawn-time = "05:00-06:00";
+          dusk-time = "22:00-23:00";
+          temp-night = 3000;
+        };
+      };'';
+      apply = c:
+        if !(c ? redshift.dawn-time || c ? redshift.dusk-time) && !(c ? redshift.location-provider) && options.locations.provider.isDefined then
+          c // {
+            redshift.location-provider = lcfg.provider;
```

Comment:
This clears all other `redshift.*` keys, you need to use `recursiveUpdate` instead

Reply:
And `location.provider` is always defined because it has a default, also `c ? dawn-time == c ? dusk-time` as per assertion, so this can be
```nix
c: if ! c ? redshift.dawn-time && ! c ? redshift.location-provider
  then recursiveUpdate c {
    redshift.location-provider = lcfg.provider;
  } else c
```